scripts/security-scan.sh

```
@@ -0,0 +1,493 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Function to scan main application dependencies
+scan_main_dependencies() {
+    local target="$1"
+    local head_ref="$2"
+    
+    echo "Security Scanning Started"
+    echo "Target: $target"
+    echo "PR Branch (code being scanned): $head_ref"
+
+    # Define directories to scan with their specific configurations
+    local scan_configs=(
+        "code-editor-src::root"
+        "remote::subdir_ignore_errors"
+        "extensions::subdir_ignore_errors"
+        "remote/web::subdir_ignore_errors"
+    )
+    local scan_results=()
+    
+    # Scan each directory
+    for config in "${scan_configs[@]}"; do
+        local dir=$(echo "$config" | cut -d':' -f1)
+        local scan_type=$(echo "$config" | cut -d':' -f3)
+        
+        echo "=== Scanning directory: $dir ==="
+        
+        # For the first scan (code-editor-src), we need to check the root directory
+        # For others, we need to check subdirectories within code-editor-src
+        local check_dir
+        if [ "$scan_type" = "root" ]; then
+            check_dir="$dir"
+        else
+            check_dir="code-editor-src/$dir"
+        fi
+        
+        # Check if directory exists and has package-lock.json
+        if [ ! -d "$check_dir" ]; then
+            echo "Warning: Directory $check_dir does not exist, skipping..."
+            continue
+        fi
+        
+        if [ ! -f "$check_dir/package-lock.json" ]; then
+            echo "Warning: No package-lock.json found in $check_dir, skipping..."
+            continue
+        fi
```

Are we expecting these to potentially non-existing? I think these are consistently expected to be there, except if we remove all the extensions for some reason or the upstream code is refactored.
I think a better practice is to fail hard for these cases to rather than risking that we skip a `package-lock.json` due to a directory rename.